app/presenters/screen_reader_filter_description_presenter.rb

```diff
@@ -0,0 +1,48 @@
+class ScreenReaderFilterDescriptionPresenter
+  def initialize(filters, sort_option)
+    @filters = filters
+    @sort_options = sort_option.present? ? ["sorted by #{sort_option['name']}"] : []
+  end
+
+  def present
+    text = (facets_without_tags + sort_options).compact.join(", ")
+    "<span class='visually-hidden'>#{text}</span>"
+  end
+
+private
+
+  attr_reader :filters, :sort_options
+
+  def facets_without_tags
+    description = filters.each_with_object([]) do |filter, facets_description|
+      if filter.hide_facet_tag?
+        filter_label = facet_without_tag_selected_option(filter)
+
+        if filter_label.empty?
+          filter_label = facet_without_tag_default_option(filter)
+        end
+
+        facets_description << "#{filter.preposition} #{filter_label}" unless filter_label.empty?
+      end
+    end
+
+    description.compact
+  end
+
+  def facet_without_tag_selected_option(filter)
+    filter.allowed_values.each do |allowed_value|
+      if filter.value == allowed_value['value']
+        return allowed_value['label']
+      end
+    end
+    ""
+  end
+
+  def facet_without_tag_default_option(filter)
+    default_option = filter.allowed_values
+                         &.detect { |option| option['default'] }
+    return '' if default_option.nil?
+
+    default_option.fetch('label', '')
```

Comment:
Perhaps we could do this, which is a bit clearer I think.

```
# return label of selected option, else return label of default option, else return nil
def filter_label(filter)
    selected_option = filter.allowed_values.find { |option| option['value'] == filter.value }
    return selected_option['label'] unless selected_option.nil?

    default_option = filter.allowed_values.find { |option| option['default'] }
    return default_option['label'] unless default_option.nil?

    nil
end

# in facets_without_tags
label = filter_label(filter)
facets_description << "#{filter.preposition} #{label}" unless label.nil?
```

It might even be worth adding the `filter_label` method to the facet model, what do you think? It's a bit confusing that the `filters` here are actually facets.

Reply:
I thought about adding it to the model but it felt a bit eggy, I think its six of one, AFAIK this logic only appears here so I'm minded to keep it here 🤷‍♂️